### PR TITLE
Add baseline test

### PR DESCRIPTION
**Pull request by jules2689**
Closes https://github.com/InstituteforDiseaseModeling/covasim/issues/26

*Tl;dr* Introduces Baseline Tests

Details
---
- On every test run, we'll run a few basic simulations and compare the key results against a known baseline
- If they are equal, we pass
- If they are not equal, we fail with an error message like this:  image

- The hope is that the english output will make it obvious what failed, and the easy resolution (rerun the test with `UPDATE_BASELINE=1` ) should make it seamless

### CODE REVIEW

File: .github/workflows/summary.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          replace-new-line: true
+          comment-hash: 'summary_comment'
+          edit-mode: 'replace'
+          body: |
+            While unit tests and running simulations can help us validate nothing obvious has broken, the stochastic
+            nature of this model means that we can't explicitly test. This summary will help experts understand if we need to take a closer look.
+
+            <details>
```
Comment (jules2689) [inline .github/workflows/summary.yaml:66]: Should we hide the output in a by-default collapsed `details` block, or should we make it wide open to start?

**Comment by jules2689** [comment]
image


This is failing because we can't even comment from a forked repo...

**Comment by jules2689** [comment]
I'm going to look at a service that we can run on the side to handle some of the contention in ci due to the read only nature of these tokens.

I have some ideas, such as having that app verify identity by sending it the GitHub token to use in an auth header. We will get read only there too, but that will validate that we have access. Then we can use a fancy token to post a comment, for example.

**Comment by ckerr-IDM** [comment]
Thanks @jules2689 , this looks cool! How about the following workflow?
* The current baseline saved as `tests/baseline.json`
* A test `tests/test_baseline.py` that asserts equal with this file, so the test fails if there's a mismatch
* If you want to update the baseline, you can run e.g. `python test_baseline.py --update` which will overwrite `baseline.json`, which you can commit and the diff will show up on the PR.

This way it would be up to the user to handle mismatches to get the tests to pass, rather than having a summary via CI (which many people don't read :roll_eyes: ). Thoughts?

**Comment by jules2689** [comment]
> A test tests/test_baseline.py that asserts equal with this file, so the test fails if there's a mismatch

@ckerr-IDM this was actually the original intention, however because it's a stochastic model the test will fail a lot as the results could vary slightly. The intention was for the comment to be used in review, but not be used to always pass/fail.

@cliffckerr (I think) had more to say on that topic

**Comment by cliffckerr** [comment]
@jules2689 Haha sorry for the confusion :) @ckerr-IDM is also me :laughing: I was concerned before if this was going to be a massive CSV or something and every time you commit you have thousands of lines of line conflicts, but I think one file with a few numbers to check is fine -- especially since most PRs shouldn't change the epi aspect of the model at all, so you should indeed have exact match with previous results, despite the stochasticity (yay for random number seeds!)

**Comment by jules2689** [comment]
@ckerr-IDM What do you think of it now, it's a simple baseline unit test 😄

**Comment by ckerr-IDM** [comment]
@jules2689 looking good, thank you!!

**Pull request merged**